### DISCUSSION TITLE
how to change the argocd cluster secret name with registering a new cluster via the CLI

### DISCUSSION THREAD

**Discussion by xinity** [argoproj/argo-cd#16018] [General]
hey there,

I'm trying to "forge" the name of the secret created by the ARGOCD CLI when registering a new cluster.
in my case, the secret is named by the server url of the cluster instead of the name passed by `---name`

is there a way to name the secret using what has been passed as parameters ?

**Comment by pasha-codefresh** [comment]
Hey, @xinity,

You can locate a secret with a name that starts with 'cluster-' and contains your cluster name. You can then edit it and change the name under the 'data' section like this:

```
apiVersion: v1
data:
  config:
  name: cGFzaGF0ZXN0
  server: 
kind: Secret
metadata:
  annotations:
    managed-by: argocd.argoproj.io
  creationTimestamp: "2023-10-19T11:01:21Z"
  labels:
    argocd.argoproj.io/secret-type: cluster
  name: cluster-5989e3ac0a1d5014222050d5d5995b87.gr7.us-east-1.eks.amazonaws.com-524909967
  namespace: codefresh
  resourceVersion: "109400891"
  uid: 9129cc82-3481-4b61-8baf-1b64331a4033
type: Opaque
```

**Reply by xinity** [reply]
hello @pasha-codefresh it's rather the name in the section i'd like to be able to "forge" with the argocd CLI:
```
metadata:
  annotations:
    managed-by: argocd.argoproj.io
  creationTimestamp: "2023-10-19T11:01:21Z"
  labels:
    argocd.argoproj.io/secret-type: cluster
  **name: cluster-5989e3ac0a1d5014222050d5d5995b87.gr7.us-east-1.eks.amazonaws.com-524909967**
  namespace: codefresh
```

**Reply by pasha-codefresh** [reply]
yes, but i am not sure that cli support it, but direct change works in cluster, i tested it
